doc/cookbook/namedRoutes/NamedRoutes.lhs

```diff
@@ -0,0 +1,383 @@
+# Record-based APIs: the nested records case
+
+*Available in Servant 0.19 or higher*
+
+Servant offers a very natural way of constructing APIs with nested records, called `NamedRoutes`.
+
+This cookbook explains how to implement such nested-record-based-APIs using
+`NamedRoutes` through the example of a Movie Catalog.
+If you don't need the nested aspect of the record-based API, you might want to look at [Record-based
+APIs: the simple
+case](../generic/Generic.html) cookbook
+which covers a simpler implementation in which every endpoint is on the same
+level.
+
+First, we start by constructing the domain types of our Movie Catalog.
+After, we show you how to implement the API type with the NamedRoutes records.
+Lastly, we make a Server and a Client out of the API type.
+
+However, it should be understood that this cookbook does _not_ dwell on the
+built-in servant combinators as the [Structuring APIs
+](<../structuring-apis/StructuringApis.html>) cookbook already covers that angle.
+
+## Why would I want to use `NamedRoutes` over the alternative `:<|>` operator?
```

Comment:
I think this whole section can go. Or at least be integrated into the `generic` cookbook. This has nothing to do with `NamedRoutes` per se but the record approach, so would be a great fit for the `generic` cookbook. By the time we learn about the nested case and how to handle it, it's in a way "too late" to explain basics about records vs `:<|>`-separated stuffs, I think.

("This section" = until "Boilerplate time!")